docs/madr/decisions/016-policy-array-merging.md

```diff
@@ -0,0 +1,133 @@
+# Policy array merging
+
+* Status: accepted
+
+## Context and Problem Statement
+
+[New policies](005-policy-matching.md) support merging of multiple configuration.
+While merging regular fields is pretty straightforward, it is quite difficult to pick a strategy for merging list.
+Our current strategy is to replace list from the most specific policy.
+
+For example
+```
+type: Policy
+targetRef:
+  kind: Mesh
+defaults:
+  elements:
+  - a
+```
+and
+```
+type: Policy
+targetRef:
+  kind: MeshService
+  name: backend
+defaults:
+  elements:
+  - b
+```
+would result in
+```
+defaults:
+  elements:
+  - b
+```
+
+But for some cases we would like to append list and end up with
+```
+defaults:
+  elements:
+  - a
+  - b
+```
+
+Example use cases are:
+* Composing ProxyTemplate modifications.
+* Composing policies in Kong Mesh's OPA Policy
+* Potentially composing backends in stuff like MeshTrafficLog
+
+## Considered Options
+
+* Name convention
+* Field annotation
+* Merge keyword
+
+## Pros and Cons of the Options
+
+### Name convention
+
+We could have a name convention that indicates that given array is appendable not replaceable when we merge.
+
+```
+defaults:
+  composableElements:
+  - a
+```
+
+So every time there is the field with value array that starts with `composable` we append.
+
+Other potential prefixes:
+* appendable
```

Comment:
maybe `append` as a prefix? So you'll have `appendRules`, `appendBackends`, etc

Reply:
`append` works for me as well. It's shorter and easier to type.

Reply:
I like idea of `appendBackends`, `mergeBackends`, `combineBackends` or `joinBackends`

Reply:
going with append for now. We can revisit and rename before the release if needed